File: scalagreetings/src/main/scala/code/siagile/commandargs/CommandBuilder.scala
package code.siagile.commandargs


class CommandBuilder {
   var tokens: List[Option] = List()
   
   var pOption = new String
   
   var pSource = new String
  
   def buildCommand = {
       val buildingCommand = new StringBuilder
       
       buildingCommand append "scalac"
       tokens.foreach(token => buildingCommand append token)
       buildingCommand append " "
       buildingCommand append pSource

       var command = buildingCommand toString

       command = command trim
       
       command replaceAll("\\s+", " ")
   }
   
   def source(name: String) = pSource = name
   
   def noWarnings = option("nowarn")
   def beVerbose = option("verbose")
   
   def option(name: String) = {
       val option = new Option(name)
       tokens = tokens ::: List[Option](option)
   }
}

File: trunk/cli/src/main/java/code/siagile/cli/CommandBuilder.java
package code.siagile.cli;

public class CommandBuilder {

	private final String command;
	private final Options options = new Options();;

	public CommandBuilder(String command) {
		this.command = command.trim();
	}

	public Command build() {
		Command cmd = new Command(command);
		cmd.options(options);
		return cmd;
	}

	public CommandBuilder option(String option) {
		options.add(option.trim());
		return this;
	}

}
